### ISSUE TITLE
[Expo] Execution failed for task ':react-native-libsodium:buildCMakeRelWithDebInfo[arm64-v8a]'

### ISSUE THREAD

**Issue by mariomurrent-softwaresolutions** [serenity-kit/react-native-libsodium#62]
When building the app locally I'm getting the following error:

` [RUN_GRADLEW] FAILURE: Build failed with an exception.
[RUN_GRADLEW] * What went wrong:
[RUN_GRADLEW] Execution failed for task ':react-native-libsodium:buildCMakeRelWithDebInfo[arm64-v8a]'.
[RUN_GRADLEW] > com.android.ide.common.process.ProcessException: ninja: Entering directory `/private/var/folders/zj/kppl1wn13_j6ndc84b1qzb9c0000gn/T/eas-build-local-nodejs/2d00feb0-5c83-4023-8a50-909b1b50b19b/build/node_modules/react-native-libsodium/android/.cxx/RelWithDebInfo/6j1y323g/arm64-v8a'
[RUN_GRADLEW]   [1/4] Building CXX object CMakeFiles/libsodium.dir/cpp-adapter.cpp.o
[RUN_GRADLEW]   FAILED: CMakeFiles/libsodium.dir/cpp-adapter.cpp.o 
`

Any idea what's wrong? Or am I missing something installed on my Mac?

**Comment by nikgraf** [comment]
puhhh absolutely no idea, does it work. I wonder if it's related to the last release. Can you try [1.2.3](https://github.com/serenity-kit/react-native-libsodium/releases/tag/v1.2.3)

**Comment by mariomurrent-softwaresolutions** [comment]
I already tried the other versions, no change ...
Maybe it's an issue with the current EXPO 50 SDK? Or a wrong NDK?

**Comment by mariomurrent-softwaresolutions** [comment]
Well that's strange. Do you know with which Android SDK? And NDK?

**Comment by nikgraf** [comment]
that said just realized I only did the iOS simulator test. The project doesn't support Android yet. So until I need it in the project or someone pays for it I won't make it a priority. Sorry :(

My guess is Expo50 uses a more modern React Native version than the e2e tests in here. So my first move would be to upgrade to the latest react native version and see if it works.

**Comment by mariomurrent-softwaresolutions** [comment]
Alright - yeah it is working for Expo iOS.

**Comment by nikgraf** [comment]
just to clarify: happy to accept contributions. just not going to spend half a week or trying to debug it

**Comment by landabaso** [comment]
Hi! Thanks for providing this package! I hit the same prob, also using Expo 50. Did you do find any workaround @mariomurrent-softwaresolutions ? Maybe go back to Expo 49?

**Comment by mariomurrent-softwaresolutions** [comment]
> Hi! Thanks for providing this package! I hit the same prob, also using Expo 50. Did you do find any workaround @mariomurrent-softwaresolutions ? Maybe go back to Expo 49?

Going back ist not an option, since EXPO SDK 49 will be deprecated in the future.
I searched the enable_if_t in the /node_modules/react-native/ReactCommon/jsi/jsi/jsi.h and replaced it with enable_if

That solve the build at least as a quick fix. But this is just a temporary solution.
You can patch the package and it should work, until this is fixed

**Comment by landabaso** [comment]
> > Hi! Thanks for providing this package! I hit the same prob, also using Expo 50. Did you do find any workaround @mariomurrent-softwaresolutions ? Maybe go back to Expo 49?
> 
> Going back ist not an option, since EXPO SDK 49 will be deprecated in the future. I searched the enable_if_t in the /node_modules/react-native/ReactCommon/jsi/jsi/jsi.h and replaced it with enable_if
> 
> That solve the build at least as a quick fix. But this is just a temporary solution. You can patch the package and it should work, until this is fixed

I found a better fix. See the PR.

**Comment by mariomurrent-softwaresolutions** [comment]
> > > Hi! Thanks for providing this package! I hit the same prob, also using Expo 50. Did you do find any workaround @mariomurrent-softwaresolutions ? Maybe go back to Expo 49?
> > 
> > 
> > Going back ist not an option, since EXPO SDK 49 will be deprecated in the future. I searched the enable_if_t in the /node_modules/react-native/ReactCommon/jsi/jsi/jsi.h and replaced it with enable_if
> > That solve the build at least as a quick fix. But this is just a temporary solution. You can patch the package and it should work, until this is fixed
> 
> I found a better fix. See the PR.

Thx for investigating and fixing :) Hope this PR will be merged soon

**Comment by nikgraf** [comment]
Merged it, but now e2e tests are failing for android complaining about the CMake version. I guess that's related. Anyone has time to look into it? I have earliest time in mid march :(

**Comment by landabaso** [comment]
I can build it for android using `yarn example android`(there were some warnings but compiles fine in my case) and all the tests passed in the simulator.

Do you want to fix the compilation warnings or are you talking about a different issue?
I'm not sure how to reproduce the problem.

Can you try running the tests and confirm @mariomurrent-softwaresolutions ? (you need to follow this: https://github.com/serenity-kit/react-native-libsodium/blob/main/CONTRIBUTING.md)

**Comment by nikgraf** [comment]
This script was failing on the main branch. I also realized you didn't follow the commit guidlines which makes releasing a bit more tricky.

I force pushed your commit with a new commit message and released. If one of you has time to investigate it would be great to improve see whats wrong here: https://github.com/serenity-kit/react-native-libsodium/blob/main/.github/workflows/e2e-android.yml

**Comment by mariomurrent-softwaresolutions** [comment]
@landabaso Well I tried out the new version and its actually building, but the App is not launching anymore ... so it crashes before it even comes up :(

**Comment by nikgraf** [comment]
@landabaso yeah, the e2e tests ran fine - so no issue anymore. Seems like it was a hickup in the CI that happened multiple times on this day.

The PR was canceled, because the commit hook was bypassed. So I fixed it and overwrote the commit. All good again. Thanks for identifying and fixing the issue :)

**Comment by mariomurrent-softwaresolutions** [comment]
@landabaso I don't even figured out what the exact error is ... neither iOS or Android log state very much :(

**Comment by landabaso** [comment]
> @landabaso I don't even figured out what the exact error is ... neither iOS or Android log state very much :(

You modified the sources, right? Try to delete any associated patch-package, completely uninstall the library, install it again and try. Delete the app in the simulator and reinstall the new one. Clear any caches. It should work. The only change applied was using a different compiler version (more modern).

**Comment by mariomurrent-softwaresolutions** [comment]
Yeah, I already tried this as well ...
Maybe I'll give it a try again this week
